### PR TITLE
Improved configuration of the tool

### PR DESCRIPTION
**Pull request by proAlexandr**
- No more multiple .env files. 
- Removed unnecessary env variables, such as Channel name, slack username etc, because it should be configured through the Slack UI during webhook configuration.
- Now deployment work without installed AWS CLI. You fill access key and secret at .env file
- Cloudwatch notifications don’t require specific SNS topic name to have a proper parsing.
- Makefile changed to npm scripts to keep consistent and run everything through npm.

### CODE REVIEW

**Comment by CodyReichert** [comment]
Oh this looks very exciting and very much much needed :D Thanks for the PR

Give me a couple of hours and I'll pull it and test everything out and leave any comments for you. :+1:

**Comment by lnikell** [comment]
@CodyReichert Also we wrote the article with full installation instructions, so next week it will be published and we will submit another PR with the link with a full guide.

File: index.js [hunk 1]
```diff
@@ -368,29 +358,35 @@ var processEvent = function(event, context) {
   var slackMessage = null;
   var eventSubscriptionArn = event.Records[0].EventSubscriptionArn;
   var eventSnsSubject = event.Records[0].Sns.Subject || 'no subject';
-  var eventSnsMessage = event.Records[0].Sns.Message;
+  var eventSnsMessageRaw = event.Records[0].Sns.Message;
+  var eventSnsMessage = null;
+
+  try {
```
Comment (creichert) [inline index.js]: indentation

File: README.md [hunk 2]
```diff
@@ -17,42 +11,36 @@ ways:
 
 **Better default formatting for CloudWatch notifications:**
 
-![AWS Cloud Notification for Slack](https://github.com/assertible/lambda-cloudwatch-slack/raw/master/images/cloudwatch.png)
+![AWS Cloud Notification for Slack](https://github.com/pixel-point/lambda-cloudwatch-slack/raw/master/images/cloudwatch.png)
```
Comment (creichert) [inline README.md]: Why were these changed to pixel point?

File: index.js [hunk 3]
```diff
@@ -5,17 +5,7 @@ var config = require('./config');
 var _ = require('lodash');
 var hookUrl;
 
-var baseSlackMessage = {
-  channel: config.slackChannel,
```
Comment (creichert) [inline index.js]: Are these variables removed completely? Or filled somewhere else?

File: package.json [hunk 4]
```diff
@@ -1,11 +1,12 @@
...
   "name": "lambda-cloudwatch-slack",
-  "version": "0.2.1",
+  "version": "0.3.0",
   "description": "Better Slack notifications for AWS CloudWatch",
   "authors": [
     "Assertible <admin@assertible.com>",
     "Christopher Reichert <creichert07@gmail.com>",
-    "Cody Reichert <codyreichert@gmail.com>"
+    "Cody Reichert <codyreichert@gmail.com>",
+    "Alexandr Promakh <s-promakh@ya.ru>"
```
Comment (creichert) [inline package.json]: +1. Can you do me a favor? And _remove_ assertible? I can do it in a follow-up also, no worries.

**Comment by creichert** [comment]
This is fantastic work. And I will definitely merge it. I have a few
observations:

- I use Makefiles extensively on every project. But mainly to automate
  my own commands. I think it's good to remove it as it can confuse many
  people used to simple nodejs infrastructure. :+1:

> @CodyReichert Also we wrote the article with full installation instructions, so next week it will be published and we will submit another PR with the link with a full guide. 

- Do you want us to wait to pull this in until you release your blog?
  Either way, _definitely_ open a new PR when you do.

- Can you fix the simple review comments? If you disagree with any,
  that's okay they are not blockers.


Additionally, it would be great if you and your team could help us keep
improving the per-service notifications to be even more accurate.

All-in-all, I'm very happy about this PR. Thank you!

File: index.js (hunk 3 above)
Comment (proAlexandr) [reply index.js]: This variables are completely removed. I and @lnikell think that filling variables in Slack UI during a webhook configuration is a better way to customize a bot.

File: README.md (hunk 2 above)
Comment (proAlexandr) [reply README.md]: It's mistake. I will revert this changes.

File: package.json (hunk 4 above)
Comment (proAlexandr) [reply package.json]: No problem, I will do it.

**Comment by proAlexandr** [comment]
@creichert @CodyReichert
+ Do we need to have `package-lock.json` in git? Or I can add it to `.gitignore`?
+ I prefer if PR be merged with squash to not store unnecessarily changes like 
```
- ![AWS Cloud Notification for Slack](https://github.com/assertible/lambda-cloudwatch-slack/raw/master/images/cloudwatch.png)
+ ![AWS Cloud Notification for Slack](https://github.com/pixel-point/lambda-cloudwatch-slack/raw/master/images/cloudwatch.png)
``` 
in git. Do you agree?

File: package.json (hunk 4 above)
Comment (creichert) [reply package.json]: thank you

File: index.js (hunk 3 above)
Comment (creichert) [reply index.js]: Ah, great find! I didn't realize that. I think it might be a good idea, in the future, to consider attachments for better display of some information. I suppose, we really need better parsers in general to get the info that's needed out of generic messages.

Just so it's noted: my biggest goal with formatting the messages, was to **avoid generic message title** text, like "Lambda CloudWatch Notification". So, if you notice incoming messages that lack actionable formatting, don't hesitate to pull again! :)

**Comment by creichert** [comment]
@proAlexandr 	I can rebase them, but in the future you could do an interactive rebase locally:

```
git rebase -i HEAD~6 # rebase number of commits on that base branch
```

Then, squash all commits into the last one and `git push -f origin branch`. No worries, though, just saying that is possible.

**Comment by creichert** [comment]
@proAlexandr fantastic contribution, I can add you as a collaborator on this repo so you can more quickly make changes and approve pull requests, if you want? We'd still like to have a PR for all changes.

**Comment by proAlexandr** [comment]
> I can add you as a collaborator on this repo so you can more quickly make changes and approve pull requests, if you want?

@creichert Thank you, but for now, I think it is not necessary.
